File: packages/clients/varity-client-go/thirdweb/chains.go
package thirdweb

// Predefined chain configurations

// VarityL3Testnet is the Varity L3 Testnet configuration
var VarityL3Testnet = ChainConfig{
	ChainID: 33529,
	Name:    "Varity L3 Testnet",
	NativeCurrency: Currency{
		Name:     "USDC",
		Symbol:   "USDC",
		Decimals: 6,
	},
	RPCUrls: []string{
		"https://rpc-varity-l3-testnet-33529-czcj67dz6a.t.conduit.xyz",
	},
	BlockExplorerUrls: []string{
		"https://explorerl2new-varity-l3-testnet-33529-czcj67dz6a.t.conduit.xyz",
	},
	Testnet: true,
}

// ArbitrumSepolia is the Arbitrum Sepolia testnet configuration
var ArbitrumSepolia = ChainConfig{
	ChainID: 421614,
	Name:    "Arbitrum Sepolia",
	NativeCurrency: Currency{
		Name:     "Ethereum",
		Symbol:   "ETH",
		Decimals: 18,
	},
	RPCUrls: []string{
		"https://sepolia-rollup.arbitrum.io/rpc",
	},
	BlockExplorerUrls: []string{
		"https://sepolia.arbiscan.io",
	},
	Testnet: true,
}

// GetChainByID returns a chain configuration by chain ID
func GetChainByID(chainID int64) *ChainConfig {
	switch chainID {
	case 33529:
		return &VarityL3Testnet
	case 421614:
		return &ArbitrumSepolia
	default:
		return nil
	}
}

// GetChainByName returns a chain configuration by name
func GetChainByName(name string) *ChainConfig {
	switch name {
	case "varity-l3-testnet", "varity", "varity-testnet":
		return &VarityL3Testnet
	case "arbitrum-sepolia", "arb-sepolia":
		return &ArbitrumSepolia
	default:
		return nil
	}
}

// SupportedChains returns all supported chain configurations
func SupportedChains() []ChainConfig {
	return []ChainConfig{
		VarityL3Testnet,
		ArbitrumSepolia,
	}
}

File: packages/clients/varity-client-python/varity_client/chains.py
"""
Varity Chain Configurations

Chain configuration dataclasses for Varity L3 and related networks.
"""

from dataclasses import dataclass
from typing import Optional


@dataclass
class Chain:
    """
    Blockchain network configuration.

    Attributes:
        chain_id: Network chain ID
        name: Human-readable network name
        rpc_url: Primary RPC endpoint URL
        rpc_urls: List of fallback RPC URLs
        explorer_url: Block explorer URL
        native_currency_name: Name of native currency
        native_currency_symbol: Symbol of native currency
        native_currency_decimals: Decimals for native currency
        is_testnet: Whether this is a testnet
    """

    chain_id: int
    name: str
    rpc_url: str
    rpc_urls: list[str]
    explorer_url: str
    native_currency_name: str
    native_currency_symbol: str
    native_currency_decimals: int
    is_testnet: bool = True


# Varity L3 Testnet Configuration (Chain ID: 33529)
VARITY_L3_TESTNET = Chain(
    chain_id=33529,
    name="Varity L3 Testnet",
    rpc_url="https://varity-l3-testnet.rpc.caldera.xyz/http",
    rpc_urls=[
        "https://varity-l3-testnet.rpc.caldera.xyz/http",
        "https://varity-l3-testnet.rpc.caldera.xyz/ws",
    ],
    explorer_url="https://varity-l3-testnet.explorer.caldera.xyz",
    native_currency_name="USDC",
    native_currency_symbol="USDC",
    native_currency_decimals=6,  # USDC has 6 decimals
    is_testnet=True,
)

# Arbitrum Sepolia (Testnet)
ARBITRUM_SEPOLIA = Chain(
    chain_id=421614,
    name="Arbitrum Sepolia",
    rpc_url="https://sepolia-rollup.arbitrum.io/rpc",
    rpc_urls=[
        "https://sepolia-rollup.arbitrum.io/rpc",
        "https://arbitrum-sepolia.publicnode.com",
    ],
    explorer_url="https://sepolia.arbiscan.io",
    native_currency_name="Sepolia Ether",
    native_currency_symbol="ETH",
    native_currency_decimals=18,
    is_testnet=True,
)

# Arbitrum One (Mainnet)
ARBITRUM_ONE = Chain(
    chain_id=42161,
    name="Arbitrum One",
    rpc_url="https://arb1.arbitrum.io/rpc",
    rpc_urls=[
        "https://arb1.arbitrum.io/rpc",
        "https://arbitrum-one.publicnode.com",
    ],
    explorer_url="https://arbiscan.io",
    native_currency_name="Ether",
    native_currency_symbol="ETH",
    native_currency_decimals=18,
    is_testnet=False,
)


def get_chain_by_id(chain_id: int) -> Optional[Chain]:
    """
    Get chain configuration by chain ID.

    Args:
        chain_id: Network chain ID

    Returns:
        Chain configuration or None if not found
    """
    chains = {
        33529: VARITY_L3_TESTNET,
        421614: ARBITRUM_SEPOLIA,
        42161: ARBITRUM_ONE,
    }
    return chains.get(chain_id)
